assimilation_code/programs/obs_sequence_tool/obs_sequence_tool.f90

```diff
@@ -741,6 +742,9 @@ program obs_sequence_tool
    print*, 'Total number of selected obs in all files :', get_num_key_range(seq_out)
 endif
 
+!setting flag to output precomputed FOs based on namelist
+call set_obs_seq_precomputed_FOs(seq_out,write_external_FOs)
```

Comment:
with the team's approval, i'd like to leave write_external_FOs as a local logical but remove it from the namelist and add one of these two: 

discard_precomputed_FOs or 
remove_precomputed_FOs 

as string arrays, before this gets entrenched in too many namelists.   the initial values are "" (empty string) and the only valid value accepted is "all" for now.  if it's "", write_external_FOs stays true (should be the default value) and if it's "all" then it changes to false.  i think then the rest of the code works as-is.

at some future point someone can add support for getting the index values from a list of obs_type names and turning them off individually.  but this is sufficient for now, i'd think.

Reply:
What I implemented supports getting index values for a list of obs_type names and turns the write flag ON for them.
You can check out the branch I supplied and compare those differences to any branch you like with the following syntax:
```
git fetch upstream
git branch -b tims_crazy_branch upstream/criedel40-write_FOs_seq_tool
git diff upstream/master
```

If you want more surgical control, you can do the following:
```
git diff upstream/master -- obs_sequence_tool.f90
```

If you have the 'gitdiff' alias, you can replace `git diff` with `gitdiff` and it whips up the differences in diffuse.

The present default behavior is to suppress writing out the precomputed FOs (i.e. they are already discarded or removed), so I think the namelist variable should have the opposite implication: preserve_precomputed_FOs

Reply:
i actually think the default behavior for this tool should be to default to preserving the info, and letting the user choose to discard it if they have a specific need.